Add checkouts api

vladanost opened the pull request:
It looks a bit strange with Cart/Cart1 and Tax/Tax1 for example

jbrinley reviewed (commented):
1. It looks like there are conflicts between the Checkout domain and some of the other domains that might need to be manually resolved. E.g., a Cart line item and a Checkout line item may not have the same schema. To resolve, we may need to modify some of the names in the yml file to, e.g., "CheckoutLineItem".
2. This created five different "*Api.php" files, when there should only be one for checkout. Look at the "tags" properties in the schema, and ensure that they all use "Checkout". That might resolve Tax/Tax1 issue, too, but I'm not 100% sure.

vladanost commented:
Correcting the Tag in the checkout.yaml resolved the issue with multiple apis.
I moved the checkout in the codegen.sh to the front of the list. That way cart.yaml that comes after it makes final changes to the Cart model.
For duplicate models like Cart/Cart1, Tax/Tax1, Currency... I added $refs to the checkout.yaml. I added the $ref for the cart from the cart.yaml - not sure if this is the right thing to do?

jbrinley reviewed (approved):
Yup, this looks a lot better. Thanks, @vladanost!
